ball.js

```diff
@@ -0,0 +1,147 @@
+//https://maththebeautiful.com/ball-bounce/
+
+class Ball
+{
+	constructor()
+	{
+		this.pos = new Position();
+		this.radius = 0;
+		this.startAngle = 0;
+		this.endAngle = 2 * Math.PI;
+		this.game = undefined;
+		this.speedX = 0;
+		this.speedY = 0;
+		this.angle = 0;
+		this.failedAudio = document.querySelector('#fail');
+		this.touchAudio = document.querySelector('#touch');
+		this.moveDirection = undefined;
+		this.firstSetDirection = 1;
+		// this.maxbounceAngle = 0;
+		this.maxAngle = 0;
+		this.maxSpeed = 0;
+		this.init = () =>
+		{
+			this.pos.x = this.game.board.dim.width / 2;
+			this.pos.y = this.game.board.dim.height / 2;
+			this.radius = this.game.board.dim.width * 0.012;
+			this.speedX = (this.radius / 2) * 0.5;
+			// this.speedY = (this.radius / 2) * 0.5;
+			this.maxAngle = this.degrees_to_radians(75);
+			if (this.firstSetDirection == 1)
+			{
+				let direction = Math.floor(Math.random() * 10);	
+				if (direction % 2 == 0)
+					this.angle = this.degrees_to_radians(180);
+				else
+					this.angle = this.radians_to_degrees(0);
+			}
+			this.firstSetDirection = 0;
+		}
+        this.update = () =>
+        {
+			this.radius = this.game.board.dim.width * 0.012;
+		}
+		this.render = () =>
+		{
+			this.radius = this.game.board.dim.width * 0.012;
+			this.move();
+			this.game.board.ctx.beginPath();
+			this.game.board.ctx.arc(this.pos.x, this.pos.y, this.radius, this.startAngle, this.endAngle);
+			this.game.board.ctx.fill();
+		}
+		this.move = () =>
+		{
+			if (this.game.continueAnimating == true)
+			{
+				let newPosX = this.pos.x + Math.cos(this.angle) * this.speedX;
+				let newPosY = this.pos.y + Math.sin ( this.angle ) * this.speedX;
+				if (this.game.player_one.isLeftPlayer(newPosX, newPosY) == true)
+				{
+     				this.angle = Math.PI - this.angle;
+					this.angle = this.degrees_to_radians(0);
+					this.angle -= this.degrees_to_radians(this.getBounceAngle());
+				}
+				else if (this.game.player_two.isRightPlayer(newPosX, newPosY) == true) 
+				{
+					this.pos.x = this.game.player_two.pos.x - this.radius;
+     				this.angle = Math.PI - this.angle;
+					this.angle += this.degrees_to_radians(this.getBounceAngle());
+				}
+				else
+				{
+					this.pos.x += Math.cos (this.angle) * this.speedX;
+					this.pos.y += Math.sin (this.angle) * this.speedX;
+				}
+				this.wasTopWallHit();
```

Comment:
Question : is audio inside render ?

Reply:
It was inside this.move() function that was itself inside render but since I have moved it in update following your recommendations, it is not inside render anymore.